cmd/utils.go

```diff
@@ -30,8 +30,17 @@ func printChain(slice []string) {
 	fmt.Println(strings.Join(slice, " -> "))
 }
 
-func getDepInfo() (map[string][]string, []string, string) {
+// DependencyOverview holds dependency module informations
+type DependencyOverview struct {
+	// Dependency graph edges modelled as node plus adjacency nodes
+	Graph map[string][]string
+	// List of all (including transitive) dependencies
+	DepList []string
+	// Name of the module from which the dependencies are computed
+	MainModuleName string
+}
 
+func getDepInfo() *DependencyOverview {
```

Comment:
Since we never modify the value returned by this function, shouldn't we just return `DependencyOverview` instead of a pointer? 🤔

Reply:
the motivation here is to not copy the data every time this is passed somewhere as the struct can be pretty heavy I think (lots of strings for big projects)

Reply:
Oh I understand now. Thanks!